packages/algob/test/lib/ssc.ts

```diff
@@ -11,20 +13,113 @@ describe("Convert integer to big endian", () => {
    * fmt.Println(buf)
    */
   it("should return correct big endian for given integers", () => {
-    let res = intToBigEndian(20);
-    let expected = new Uint8Array([0, 0, 0, 0, 0, 0, 0, 20]);
+    let res = uint64ToBigEndian(0);
+    let expected = new Uint8Array([0, 0, 0, 0, 0, 0, 0, 0]);
     assert.deepEqual(res, expected);
 
-    res = intToBigEndian(700000000);
+    res = uint64ToBigEndian(20);
+    expected = new Uint8Array([0, 0, 0, 0, 0, 0, 0, 20]);
+    assert.deepEqual(res, expected);
+
+    res = uint64ToBigEndian(700000000);
     expected = new Uint8Array([0, 0, 0, 0, 41, 185, 39, 0]);
     assert.deepEqual(res, expected);
 
-    res = intToBigEndian(233654);
+    res = uint64ToBigEndian(233654);
     expected = new Uint8Array([0, 0, 0, 0, 0, 3, 144, 182]);
     assert.deepEqual(res, expected);
 
-    res = intToBigEndian(9007199254740991);
+    res = uint64ToBigEndian(Number.MAX_SAFE_INTEGER);
     expected = new Uint8Array([0, 31, 255, 255, 255, 255, 255, 255]);
     assert.deepEqual(res, expected);
   });
+
+  it("should return correct big endian for 64 bit integer", () => {
+    let res = uint64ToBigEndian(MIN_UINT64);
+    let expected = new Uint8Array([0, 0, 0, 0, 0, 0, 0, 0]);
+    assert.deepEqual(res, expected);
+
+    res = uint64ToBigEndian(233654n);
+    expected = new Uint8Array([0, 0, 0, 0, 0, 3, 144, 182]);
+    assert.deepEqual(res, expected);
+
+    res = uint64ToBigEndian(18446744073709551055n);
+    expected = new Uint8Array([255, 255, 255, 255, 255, 255, 253, 207]);
+    assert.deepEqual(res, expected);
+
+    res = uint64ToBigEndian(18446744073709111055n);
+    expected = new Uint8Array([255, 255, 255, 255, 255, 249, 71, 15]);
+    assert.deepEqual(res, expected);
+
+    res = uint64ToBigEndian(MAX_UINT64);
+    expected = new Uint8Array([255, 255, 255, 255, 255, 255, 255, 255]);
+    assert.deepEqual(res, expected);
+  });
+
+  it("should throw error if value is not uint64", () => {
+    let errMsg = "Invalid uint64 -5";
+    assert.throws(() => uint64ToBigEndian(MIN_UINT64 - 5n), errMsg);
+
+    errMsg = "Invalid uint64 18446744073709551620";
+    assert.throws(() => uint64ToBigEndian(MAX_UINT64 + 5n), errMsg);
+  });
+});
+
+describe("Parse appArgs to SSC to bytes", () => {
+  it("should return undefined if app Args are not defined", () => {
+    const res = parseSSCAppArgs(undefined);
+    assert.isUndefined(res);
+  });
+
+  it("should return same bytes if all bytes are passed", () => {
+    const res = parseSSCAppArgs(['a', 'b', 'c'].map(stringToBytes));
```

Comment:
can we change it to numbers(like in line below) ?

Reply:
or not this seems non-intutive what the next line is doing.

Reply:
Can you please elaborate? In `res` i am using passing string in arrays and using `stringToBytes` because that's probably how the user will pass `appArgs`.
`expected` is the expected value from the parse function, since this was expected i added actual `byte` array here instead of using the `stringToBytes` function.

Reply:
a, b, c are 3 different strings, in case of `int` test(the next unit test) we have three [] arrays, but here we have 3 values in one array. that seems un-intuitive

Reply:
Here we are creating the array inside map function. I thought that was understood. Will update this as per next unit test.